### PR TITLE
WIP: experiments with the Glucose SAT solver

### PR DESCRIPTION
**Pull request by mcg1969**
cc: @kalefranz @jjhelmus 

On a whim I decided to see how hard it would be to use conda with an alternative to pycosat. The biggest challenge is likely going to be getting something that can be compiled as a C extension---something that I do suspect will ultimately be necessary. But for this experiment, I chose to call the SAT solver as a subprocess.

The solver I chose was [Glucose](http://www.labri.fr/perso/lsimon/glucose/), a solver based on [MiniSAT](http://minisat.se) and a rather frequent winner of SAT competitions. 

To try this out, you'll need the `glucose` solver in your environment. Either compile it by hand or use by [conda package](https://anaconda.org/mcg/glucose) (Mac only for the moment).

Bottom line is: it works, it has a lot of subprocess overhead. The likelihood that I'll be able to work on this a lot is small. But note the _very_ small diff required to get this going, thanks to the way I refactored solve/logic way back when. So perhaps this will encourage a conda developer to take this on.

### CODE REVIEW

**Comment by mcg1969** [comment]
Another alternative is [CryptoMiniSat](https://github.com/msoos/cryptominisat), which as of version 5.0.1 has an MIT license and already has a Python interface.

**Comment by mbargull** [comment]
Closing in favor of gh-7676. With those changes, we can play with Glucose (via the `python-sat` package on PyPi) and CryptoMiniSat.
